.github/workflows/release-and-update-docker.yml

```diff
@@ -0,0 +1,143 @@
+name: Auto Release on Main
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+    steps:
+      - name: Checkout repository with full history
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Required for go-semantic-release
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.22'
+
+      - name: Run Go linter
+        uses: golangci/golangci-lint-action@v3
+        with:
+          version: latest
+
+      - name: Run Go tests
+        run: go test -v ./...
+
+      - name: Generate commit summary
+        id: git-summary
+        run: |
+          echo "## Summary" >> commit-summary.md
+          echo "" >> commit-summary.md
+          # Count different types of changes
+          FEATURES=$(git log $(git describe --tags --abbrev=0)..HEAD --pretty=format:"%s" --no-merges | grep -i "^feat" | wc -l)
+          FIXES=$(git log $(git describe --tags --abbrev=0)..HEAD --pretty=format:"%s" --no-merges | grep -i "^fix" | wc -l)
+          DOCS=$(git log $(git describe --tags --abbrev=0)..HEAD --pretty=format:"%s" --no-merges | grep -i "^docs" | wc -l)
+          echo "* $FEATURES new features" >> commit-summary.md
+          echo "* $FIXES bug fixes" >> commit-summary.md
+          echo "* $DOCS documentation updates" >> commit-summary.md
+          echo "" >> commit-summary.md
+          echo "## Changes since last release" >> commit-summary.md
+          echo "" >> commit-summary.md
+          git log $(git describe --tags --abbrev=0)..HEAD --pretty=format:"* %s (%h)" --no-merges >> commit-summary.md
+          echo "COMMIT_SUMMARY<<EOF" >> $GITHUB_ENV
+          cat commit-summary.md >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
+
+      - name: Run semantic versioning and release
+        id: semantic-release
+        uses: go-semantic-release/action@v1
```

Comment:
The `go-semantic-release` library will replace the existing tagging process offered by the Go native method. It enforces version format, auto-bump versions, and create a release.

The scripts are not tested yet, but if the plan in RELEASE.md makes sense I will test and finalize them.